kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -191,13 +191,43 @@ public static Target asTarget(URI uri) {
   public static <E> PCollection<E> partition(PCollection<E> collection,
                                              View<E> view,
                                              int numWriters) {
+    return partition(collection, view, numWriters, 1);
+  }
+
+  /**
+   * Partitions {@code collection} to be stored efficiently in {@code View}.
+   * <p>
+   * This restructures the parallel collection so that all of the entities that
+   * will be stored in a given partition will be evenly distributed across a specified
+   * {@code numPartitionWriters}.
+   * <p>
+   * If the dataset is not partitioned, then this will structure all of the
+   * entities to produce a number of files equal to {@code numWriters}.
+   *
+   * @param collection a collection of entities
+   * @param view a {@link View} of a dataset to partition the collection for
+   * @param numWriters the number of writers that should be used
+   * @param numPartitionWriters the number of writers data for a single partition will be distributed across
+   * @param <E> the type of entities in the collection and underlying dataset
+   * @return an equivalent collection of entities partitioned for the view
+   * @see #partition(PCollection, View)
+   *
+   * @since 1.1.0
+   */
+  public static <E> PCollection<E> partition(PCollection<E> collection,
+                                             View<E> view,
+                                             int numWriters, int numPartitionWriters) {
+
+    //ensure the number of writers is honored whether it is per partition or total.
+    int writers = Math.max(numPartitionWriters, numWriters);
```

Comment:
Not, not really. Those 10 reducers will run through multiple tasks. With 15 tasks per partition, each one will get at least one task. Each task will create a writer, write the data, and close that writer, creating one file in one partition (because we know all the data is for the same partition).